control/actions.cpp

```diff
@@ -285,6 +285,86 @@ Action set_ceiling_from_average_sensor_value(
     };
 }
 
+Action conditionally_set_floor_from_average_sensor_value(
+        std::vector<PrecondGroup>&& pg,
+        std::map<int64_t, uint64_t>&& val_to_speed)
+{
+    return [pg = std::move(pg),
+            val_to_speed = std::move(val_to_speed)](control::Zone& zone,
+                                                    const Group& group)
+    {
+        auto preGrp = std::all_of(
+            pg.begin(),
+            pg.end(),
+            [&zone](auto const& entry)
+            {
+                try
+                {
+                    return zone.getPropValueVariant(
+                        std::get<pcPathPos>(entry),
+                        std::get<pcIntfPos>(entry),
+                        std::get<pcPropPos>(entry)) ==
+                                std::get<pcValuePos>(entry);
+                }
+                catch (const std::out_of_range& oore)
+                {
+                    // Default to property variants not equal when not found
+                    return false;
+                }
+            });
+
+        if (!preGrp)
+        {
+            // Do not update the floor speed
+            return;
+        }
+
+        auto speed = zone.getDefFloor();
```

Comment:
maybe when you upstream you could add some more comments on what the code below is doing because i'm having a hard time seeing how it applies what the function describes.

Reply:
So apparently when you resolve a conversation, you lose the comment I had written.

The code below is actually the same as the current set floor action function above(set_floor_from_average_sensor_value). This action just adds a preconditional group that is used to determine if the floor table provided should be used on this zone.

This is an action that may be handled upstream with proper event configuration and therefore wont be included when pushed upstream if that's the case. However, if it is necessary to push upstream as-is, I will include additional comments on what this is doing.